geode-core/src/main/java/org/apache/geode/cache/client/internal/QueueManagerImpl.java

```diff
@@ -966,6 +965,16 @@ public void recoverPrimary(Set<ServerLocation> excludedServers) {
     }
   }
 
+  private boolean isPrimaryRecoveryNeeded(final ConnectionList queueConnectionList) {
```

Comment:
If you make this static you could easily create unit tests that could test all the paths in this method.

Reply:
@pivotal-jbarrett done, all branches are covered now.